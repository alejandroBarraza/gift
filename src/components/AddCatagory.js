import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCatagory = ({ setcategories }) => {
    const [inputValue, setinputValue] = useState('');

    //function update input text
    const handleText = ({ target: { value } }) => {
        setinputValue(value);
    };

    //function update input value in the list
    const handleSubmit = (e) => {
        e.preventDefault();
        setcategories((cat) => [inputValue,...cat]);
        setinputValue('');
    };

    return (
        <form onSubmit={handleSubmit}>
            <h2>Add a new category</h2>
            <input type='text' required value={inputValue} onChange={handleText} />
        </form>
    );
};

AddCatagory.propTypes = {
    setcategories: PropTypes.func.isRequired,
};
