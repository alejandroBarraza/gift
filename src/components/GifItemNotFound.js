import React from 'react'

export const GifItemNotFound = ({category}) => {
    return (
        <div>
            <h2>{category} no encontrada, </h2>
            
        </div>
    )
}
