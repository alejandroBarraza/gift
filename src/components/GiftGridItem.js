import React from 'react';
import PropTypes from 'prop-types';

export const GiftGridItem = ({ url, title }) => {
    return (
        <div className='card glassmorphismo'>
            <p>{title}</p>
            <img src={url} alt={title} />
        </div>
    );
};

GiftGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
};
