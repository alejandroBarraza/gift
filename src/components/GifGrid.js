
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';

import { GiftGridItem } from './GiftGridItem';



export const GifGrid = ({ category }) => {


    const {data:imgs} = useFetchGifs(category)
    

    return (
        <>
            <h2>busqueda ralizada : {category}</h2>
           
            <div className='card-container'>
                {
                    imgs.map( (img) => <GiftGridItem key={img.id} {...img} />)
                }   
            </div>
        </>
    );
};
