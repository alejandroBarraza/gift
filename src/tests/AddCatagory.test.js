import '@wojtekmaj/enzyme-adapter-react-17';
import '../setupTests';
import { shallow } from 'enzyme';
import { AddCatagory } from '../components/AddCatagory';

describe('testing <AddCategory setCategories={setCategories} /> component', () => {
    let wrapper;
    const setcategories = jest.fn()

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCatagory setcategories={setcategories} />);
    }),
    afterEach(() => {
        wrapper.unmount();
    });


    test('<AddCategory setCategories={setCategories}/> should match snapShot()', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should check if value change in input ', () => {
        const input = wrapper.find('input');
        const value = 'hola como estas'
        input.simulate('change',{target:{value}})
        //check if value change in input
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(value);
    })

    test('check after sent submit, setcategories have been called.', () => {
        const form = wrapper.find('form');
        form.simulate('submit',{preventDefault(){}})
        expect(setcategories).toHaveBeenCalled()
    
        
    })
    test('should check if after submit setcategories has benn called, and input value = " " ', () => {
        //debemos seleccionar el input value
        //debemos simular el input value mandandole un valor
        //luego debemos simular un evento submit
        //revisar si setcategories fue llamado
        //revisar si input value tiene la casilla vacia


        const value = "test"
        wrapper.find('input').simulate('change', {target:{value:value}})

        //simular evenyto submit
        wrapper.find('form').simulate('submit',{preventDefault(){}})
        expect(setcategories).toHaveBeenCalledWith(expect.any(Function));
        //check if input value has empty value after setcagories has ben called.
        
        expect(wrapper.find('input').prop('value')).toBe('')





        // const p = wrapper.find('p');
        // expect(p.text().trim()).toBe(value);
        // console.log(p.html())
        
    })
    
    
    
    



    
})
