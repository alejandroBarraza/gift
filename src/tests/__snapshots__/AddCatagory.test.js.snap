// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <AddCategory setCategories={setCategories} /> component <AddCategory setCategories={setCategories}/> should match snapShot() 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add a new category
  </h2>
  <input
    onChange={[Function]}
    required={true}
    type="text"
    value=""
  />
</form>
`;
